### DISCUSSION TITLE
How can I create an indexed function in SymPy?

### DISCUSSION THREAD

**Discussion by Inmortak** [sympy/sympy#26188] [Q&A]
What I want is to create an indexed function. Something of the form:

$f_i(x)$

Where i is an integer and x is real. The idea is then to be able to integrate or take the x derivative:

$\int \sum_i f_i(x) dx$

I've been looking everywhere how to do it, unsuccessfully.

For example, I attempted the following code:
```python
import sympy as sy

x, n = sy.symbols('x n', real=True, integer=True)
f = sy.Function('f')
f_n = sy.Indexed(f(x), n)
f_n
```

Only to obtain the following error:

```python
TypeError: 
The base can only be replaced with a string, Symbol, IndexedBase or an
object with a method for getting items (i.e. an object with a
`__getitem__` method).
```

I've tried the next alternative:

```python
import sympy as sy

x, n = sy.symbols('x n', real=True, integer=True)
f = sy.IndexedBase(sy.Function('f'))
f_n = sy.Indexed(f, n)
f_n
```

But then I get this error:

```
SympifyError: SympifyError: f
```

Any help is greatly appreciated!

**Comment by ThePauliPrinciple** [comment]
I don't think it is currently possible to do what you are asking for.

What you could do instead is simply have `i` be a variable of the function: `f(i, x)`.

Something like:
```python
import sympy as sp
x, i = sp.symbols('x i')
f = sp.Function('f')
sp.integrate( sp.Sum(f(i,x), (i, 0, sp.symbols('N')) ), x)
```

**Comment by sylee957** [comment]
I also think that everything is possible to do with multivariate functions.
